content/en/user-guide/lambda-tools/debugging/index.md

```diff
@@ -418,6 +419,104 @@ $ awslocal lambda invoke --function-name func1 \
 {{% /tab %}}
 {{< /tabpane >}}
 
+## Lambda Debug Mode (Preview)
+
+Lambda Debug Mode is a preview feature in LocalStack designed to enhance your Lambda debugging workflows.
+This feature provides an optimized environment for debugging Lambda functions, ensuring that you have the
+necessary tools and flexibility to troubleshoot effectively.
+
+### Key Features
+* **Automatic Timeout Management**: Integrates with API Gateway to prevent Lambda function timeouts,
+giving developers ample time to connect remote debuggers and inspect the function's behavior.
+* **Multi-Function Debugging**: Supports debugging multiple Lambda functions concurrently.
+
+### Enabling Lambda Debug Mode
+
+To enable Lambda Debug Mode, set the `LAMBDA_DEBUG_MODE` environment variable as shown below:
+
+```bash
+$ LAMBDA_DEBUG_MODE=1 \
+  LAMBDA_DOCKER_FLAGS='-p 19891:19891' \
+  localstack start
+```
+
+When enabled, Lambda Debug Mode automatically adjusts timeouts to accommodate debugging needs:
+* **Lambda Container Startup Timeout**: Provides additional time for debugger connection during container creation.
+* **Lambda Execution Timeout**: Extends the execution window, allowing for in-depth remote debugging.
+* **API Gateway-Lambda Integration Timeout**: Increases timeout settings to avoid premature terminations.
```

Comment:
just a clarification question: Are API Gateway timeouts enforced or not? (remembering a discussion that they are not 🤔 ) Does it differ for apigateway v1 vs. v2?

Reply:
Technically in LocalStack it would not be API Gateway the one directly responsible for the timeout, but the executor endpoint. Originally we had the Lambda max timeout set by default (or about that, 905 seconds). With LDM we increase this too.
I decided to phrase it like so, because from a end-user perspective, it looks like API Gateway is not timing out, and I would like them to feel confident that that is not going to be an issue and automatically taken care of when debugging.